### PR TITLE
Remove TopCoder exception helper library

### PR DESCRIPTION
**Pull request by jasonaowen**
The TopCoder `BaseException` class and its related permutations allow for storing arbitrary data alongside an exception. Several of the exception classes in the PSM extended these TopCoder helper classes to get that additional functionality, but the functionality was never used.

Remove the unused constructors that allowed for that additional functionality from our exception classes, replace the base classes of our exception classes with the standard Java versions (which the TopCoder exception classes themselves extended), and then delete the now-unused TopCoder jar file.

---

I tested this by compiling (which I think would catch any potential errors, as they would all be static), deploying, and running the integration tests.

---

Issue #214 Remove TopCoder-specific jarfiles

### CODE REVIEW

File: psm-app/services/src/main/java/gov/medicaid/services/PortalServiceRuntimeException.java [hunk 1]
```diff
@@ -48,29 +45,4 @@ public PortalServiceRuntimeException(String message) {
     public PortalServiceRuntimeException(String message, Throwable cause) {
         super(message, cause);
     }
-
```
Comment (cecilia-donnelly) [inline psm-app/services/src/main/java/gov/medicaid/services/PortalServiceRuntimeException.java]: So in  calls like `throw new PortalServiceRuntimeException("Cannot verify linked practice.", e);`, `e` is the cause, not the data?  And how do you know?

**Comment by jasonaowen** [comment]
@cecilia-donnelly IntelliJ has a built-in decompiler that will show the jar's bytecode in human-readable format:

decompile

Using that, I was able to view the decompiled `BaseCriticalException`, see that it derives from `BaseException`, and see in turn that `BaseException` derives from `Exception`.

File: psm-app/services/src/main/java/gov/medicaid/services/PortalServiceRuntimeException.java (hunk 1 above)
Comment (jasonaowen) [reply psm-app/services/src/main/java/gov/medicaid/services/PortalServiceRuntimeException.java]: Right! The data would be attached as an instance of `ExceptionData`, which is one of the classes in `base_exception.jar`. The various `Exception` helper classes accept that `ExceptionData` in some of their constructors, but we never construct an instance of `ExceptionData`; we only ever refer to it in our exception classes that derive from the ones in `base_exception.jar`. I first used IntelliJ's Find Usages action, examined all the instances, found them unused, deleted them, and then did a `git grep` to confirm.

**Comment by cecilia-donnelly** [comment]
Thanks for the explanation, @jasonaowen!

**Pull request merged**